### ISSUE TITLE
revise all docstrings

### ISSUE THREAD

**Issue by nden** [spacetelescope/jwst#2681]
As part of the 7.2 documentation update make sure code has docstrings, they are correct and they are displayed correctly. Docstrings must be present in all public classes and functions, and in complex private functions/methods. 

[An example module](https://gist.github.com/nden/d8a7617424ecf9d67aaf4f485d0f561f)

Some useful links for reference:

[Numpydoc docstring convention](https://numpydoc.readthedocs.io/en/latest/format.html)
[STScI's Python style guide](https://github.com/spacetelescope/style-guides/blob/master/guides/python.md)

To build the documentation locally:

`python setup.py build_sphinx`

I had to pip install `sphinx_automodapi`, all other packages install with conda.

**Comment by stscicrawford** [comment]
List of subpackages
- [ ] ami
- [ ] assign_wcs
- [ ] associations
- [ ] background
- [ ] barshadow
- [ ] combine_1d
- [ ] coron
- [ ] csv_tools
- [ ] cube_build
- [ ] cube_skymatch
- [ ] dark_current
- [ ] datamodels
- [ ] dq_init
- [ ] emission
- [ ] engdblog
- [ ] exp_to_source
- [ ] extract_1d
- [ ] extract_2d
- [ ] firstframe
- [ ] fits_generator
- [ ] flatfield
- [ ] fringe
- [ ] gain_scale
- [ ] group_scale
- [ ] guider_cds
- [ ] imprint
- [ ] ipc
- [ ] jump
- [ ] jwpsf
- [ ] lastframe
- [ ] lib
- [ ] linearity
- [ ] model_blender
- [ ] mrs_imatch
- [ ] msaflagopen
- [ ] outlier_detection
- [ ] pathloss
- [ ] persistence
- [ ] photom
- [ ] pipeline
- [ ] ramp_fitting
- [ ] refpix
- [ ] resample
- [ ] reset
- [ ] rscd
- [ ] saturation
- [ ] skymatch
- [ ] source_catalog
- [ ] srctype
- [ ] stpipe
- [ ] straylight
- [ ] superbias
- [ ] tests
- [ ] timeconversion
- [ ] transforms
- [x] tso_photometry
- [ ] tweakreg
- [ ] tweakreg_catalog
- [ ] wfs_combine
- [ ] white_light
- [ ] wiimatch

**Comment by philhodge** [comment]
If a function only returns None (and maybe not even explicitly), does there need to be a Returns section?

**Comment by nden** [comment]
@stscieisenhamer Yes, these are inline comments and the only way for them to show up in the docs is to write them below the attribute. Alternatively there could be an `Attributes` section. I personally prefer to keep the comment with the attribute so I don't have to scroll up and down to find the description.
The above example is meant more as a reference and less as a strict template.

**Comment by nden** [comment]
@philhodge No need to have a `Returns` section if a function returns `None`.

**Comment by philhodge** [comment]
Should we edit the list of subpackages that Steve included in a comment, to let people know which steps we're working on?

**Comment by stscicrawford** [comment]
Good idea @philhodge !  If you have started working on a package, please add your name above.

**Comment by sosey** [comment]
PR #2752  updates docstrings, though not entirely complete, for assign_wcs, transforms, datamodels, lib, and extract_2d

**Comment by melanieclarke** [comment]
Linked to epic JP-3855 for updating code style.  When that is complete, this can be closed.

**Comment by melanieclarke** [comment]
All modules now updated.
